integration_test/list_test.go

```diff
@@ -42,3 +45,122 @@ func TestKrewList(t *testing.T) {
 	}
 	// TODO(ahmetb): install multiple plugins and see if the output is sorted
 }
+
+func TestKrewListJSON(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test.Krew("install",
+		"--manifest", filepath.Join("testdata", "foo.yaml"),
+		"--archive", filepath.Join("testdata", "foo.tar.gz")).
+		RunOrFail()
+
+	expected := []byte("{\n    \"Name\": \"foo\",\n    \"Version\": \"v0.1.0\"\n}\n")
+
+	eventualList := test.WithIndex().Krew("list", "-o", "json").RunOrFailOutput()
+	if diff := cmp.Diff(eventualList, expected); diff != "" {
+		t.Fatalf("'list' output doesn't match:\n%s", diff)
+	}
+
+	Plugin := struct {
+		Name    string
+		Version string
+	}{"", ""}
+
+	err := json.Unmarshal(eventualList, &Plugin)
```

Comment:
I am bit confused here. Shouldnt this be []Plugin? Since this is list command I expect a list response.

Reply:
Only one plugin (foo) is installed in the test so I chose to just use a singular Plugin.

Reply:
hmm I think when you do `kubectl get` (list cmd) you always get a PodList resource (unless you request a specific resource by NAME); that's why I expect the genericclioptions would return us a json/yaml array here.

any ideas why it's happening?

does it automatically turn into an array if you install multiple plugins
(btw you can install multiple plugins using validPlugin1 validPlugin2 directly)

Reply:
My apologies, I miscommunicated. What's returned is a list, but since only one plugin is installed in the test, I treat it as one object rather than a list.

I'll convert the Unmarshal to an array of Plugins.

Reply:
Maybe try unmarshaling the output into `[]map[string]interface{}` where map[string]interface{} is a placeholder for Plugin object.

This way, we can (1) validate it's valid yaml/json, (2) validate it's an array. :)

Reply:
I'll swap it over.

Reply:
@JJJJJones I was wrong in my review per @corneliusweig's comment above about the return format from -o=json for lists.

It appears like -o=json for lists returns a **json stream** and not json array. So what you were doing here is correct.

But for **tests**, if you install 2 plugins, you need to do something like this:

```
j := json.NewDecoder(b)
i := 0
for err := j.Decode(); err != io.EOF; {
  i++
}
// assert i == 2 (or however many plugins you installed)
```

Reply:
I'll expand the tests with validPlugin to ensure this.

Reply:
When I run `kubectl get pod -o yaml`, I get:
```
apiVersion: v1
items: []
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```
and for json:
```
{
    "apiVersion": "v1",
    "items": [],
    "kind": "List",
    "metadata": {
        "resourceVersion": "",
        "selfLink": ""
    }
}
```

Therefore, I think we should stay in line with what k8s does and print a list of objects instead of an object stream.

Reply:
oops.. nevermind. Just saw that you changed it already :bowing_man:

Reply:
Sorry, I forgot to make a comment here.

For anyone not caught up to speed, I already swapped from a JSON stream to array.